### PR TITLE
add colon indexing

### LINKED ISSUE #215: Support indexing with colon
I don't think it is currently possible to use the colon to extract rows/ columns from a worksheet.

We,d need 
```julia
getdata(::XLSX.Worksheet, ::Colon, ::Integer)
getdata(::XLSX.Worksheet, ::Integer, ::Colon)
```

for that.

### PR DESCRIPTION
**Pull request by BeastyBlacksmith**
fix #215

### CODE REVIEW

**Comment by felipenoris** [comment]
See https://github.com/felipenoris/XLSX.jl/blob/d2f51a80fcf24b4887aeb61dae8637baef80e3d8/src/worksheet.jl#L14.

The problem with this feature is that *Worksheet Dimensions* attribute is optional (see section 18.3.1.35 - dimension, from the specs). It is not guaranteed that the software writing the workbook fills this attribute, and it may also fill it incorrectly.

Also, keep in mind that `XLSX.get_dimension` might return `nothing`.

**Comment by BeastyBlacksmith** [comment]
> and it may also fill it incorrectly.

I observed that, but in that case you usually get unwanted `missing` values, which is not too hard to handle on the user side I'd say.

**Comment by Keno** [comment]
I was also looking for this. Could we get this in?

File: src/worksheet.jl [hunk 1]
```diff
@@ -77,6 +90,25 @@ See also [`XLSX.readdata`](@ref).
 """
 getdata(ws::Worksheet, single::CellRef) = getdata(ws, getcell(ws, single))
 getdata(ws::Worksheet, row::Integer, col::Integer) = getdata(ws, CellRef(row, col))
+getdata(ws::Worksheet, row::Union{Integer,UnitRange{<:Integer}}, col::Union{Integer,UnitRange{<:Integer}}) = getdata(ws, CellRange(CellRef(first(row), first(col)), CellRef(last(row), last(col))))
+function getdata(ws::Worksheet, row::Integer, ::Colon)
```
Comment (divbyzerofordummies) [inline src/worksheet.jl:94]: I think `row` should also be Union{Integer,UnitRange{<:Integer}} as `col` below
```suggestion
function getdata(ws::Worksheet, row::Union{Integer,UnitRange{<:Integer}}, ::Colon)
```

**Comment by felipenoris** [comment]
Please, add some tests.

**Comment by felipenoris** [comment]
@BeastyBlacksmith, maybe the test should be based on Julia's behavior with matrices, like below. What do you think?

```julia
sheet2_matrix = Any[ 1 2 3 ; 4 5 6 ; 7 8 9 ]
@test sheet2[1, :] == sheet2_matrix[1, :]
@test sheet2[1:2, :] == sheet2_matrix[1:2, :]
@test sheet2[:, 2] == sheet2_matrix[:, 2]
@test sheet2[:, 2] == sheet2_matrix[:, 2]
@test sheet2[:, 2:3] == sheet2_matrix[:, 2:3]
@test sheet2[1:2, 2:3] == sheet2_matrix[1:2, 2:3]
```

**Comment by felipenoris** [comment]
Alright!

**Pull request merged**